src/Altinn.Platform/Altinn.Platform.Storage/Storage/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Altinn.Platform.Storage.Extensions
+{
+    /// <summary>
+    /// Extensions to facilitate sanitization of string values
+    /// </summary>
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Sanitize the input as a file name.
+        /// </summary>
+        /// <param name="input">The input variable to be sanitized.</param>
+        /// <param name="throwExceptionOnInvalidCharacters">Throw exception instead of replacing invalid characters with '-'.</param>
+        /// <returns>A filename cleaned of any impurities.</returns>
+        public static string AsFileName(this string input, bool throwExceptionOnInvalidCharacters = true)
+        {
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                return input;
+            }
+
+            char[] illegalFileNameCharacters = Path.GetInvalidFileNameChars();
+            if (throwExceptionOnInvalidCharacters)
+            {
+                if (illegalFileNameCharacters.Any(ic => input.Any(i => ic == i)))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(input));
+                }
+
+                return input;
+            }
+
+            return illegalFileNameCharacters.Aggregate(input, (current, c) => current.Replace(c, '-'));
```

Comment:
seems like we are replacing illegal chars with `-` instead of `_` what's the reasoning for changing the char ?

Reply:
I copied the extension method from one project to another. I can easily change either, but I do think we should use the same character in both cases. 
@alt-how Any preferences?

Reply:
No huge preference, but remember backwards compability if it has been used in other solutions.

Reply:
I changed the AsFileName methods in App and Storage to replace invalid characters with '_' as that was already use by App. I have not changed the AsFileName version used by Designer.